src/commands/deploy/deploy.ts

```diff
@@ -59,8 +59,11 @@ export async function deploy(actionContext: IActionContext, arg1?: vscode.Uri |
     }
 
     const isZipDeploy: boolean = siteConfig.scmType !== constants.ScmType.LocalGit && siteConfig !== constants.ScmType.GitHub;
+    const appSettingsClient: IAppSettingsClient = node.client;
+    const remoteSettings: WebSiteManagementModels.StringDictionary = await appSettingsClient.listApplicationSettings();
+    const isBuildDuringDeploymentAppSetting: boolean = !!remoteSettings.properties && 'SCM_DO_BUILD_DURING_DEPLOYMENT' in remoteSettings.properties;
     // only check enableScmDoBuildDuringDeploy if currentWorkspace matches the workspace being deployed as a user can "Browse" to a different project
-    if (getWorkspaceSetting<boolean>(constants.configurationSettings.showBuildDuringDeployPrompt, context.effectiveDeployFsPath)) {
+    if (!isBuildDuringDeploymentAppSetting && getWorkspaceSetting<boolean>(constants.configurationSettings.showBuildDuringDeployPrompt, context.effectiveDeployFsPath)) {
```

Comment:
Would be better if you checked the app setting stuff within this first "if" so that people who turned the VS Code setting off never even have to make the http call to get app settings